src/osdc/Objecter.h

```diff
@@ -1206,10 +1216,12 @@ class Objecter : public md_config_obs_t, public Dispatcher {
 
     bool used_replica = false;
     bool paused = false;
+    bool use_osd_epoch = false;
 
     int osd = -1;      ///< the final target osd, or -1
 
     epoch_t last_force_resend = 0;
+    epoch_t epoch; ///< Required epoch
```

Comment:
maybe we can just leave the interval checking to `assert_interval` op, and instead of using `required_interval` or `use_osd_epoch`, we can have a `force_use_osd` to specify the designated osd so object will always use it as long as it's up.